Scripts/RemediationScripts/Remediate-DisableRemoteDebuggingForAppServices.ps1

```diff
@@ -0,0 +1,879 @@
+
+            try
+            {
+                $appServiceResource = Get-AzResource -ResourceId $resourceId -ErrorAction SilentlyContinue
+                $appServiceResources += $appServiceResource
+            }
+            catch
+            {
+                Write-Host "Error fetching App Service resource: Resource ID - $($resourceId). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host "Skipping this App Service resource..." -ForegroundColor $([Constants]::MessageType.Warning)
+            }
+        }
+    }
+
+    $totalAppServices = ($appServiceResources | Measure-Object).Count
+
+    if ($totalAppServices -eq 0)
+    {
+        Write-Host "No App Services found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+  
+    Write-Host "Found $($totalAppServices) App Service(s)." -ForegroundColor $([Constants]::MessageType.Update)
+
+    # Includes App Services where Remote Debugging is disabled on all slots - production slot and all non-production slots.
+    $appServicesWithRemoteDebuggingDisabled = @()
+
+    # Includes App Services where Remote Debugging is not disabled on all slots - production slot or one or more non-production slots.
+    $appServicesWithoutRemoteDebuggingDisabled = @()
+
+    # Includes App Services that were skipped during remediation. There were errors remediating them.
+    $appServicesSkipped = @()
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "`n[Step 3 of 5] Fetching all App Service configurations..."
+
+    $appServiceResources | ForEach-Object {
+        $appServiceResource = $_
+        $resourceId = $_.ResourceId
+        $resourceGroupName = $_.ResourceGroupName
+        $resourceName = $_.ResourceName
+        
+        try
+        {
+            $nonCompliant = @()
+            $nonProductionSlots = @()
+            $isRemoteDebuggingDisabledOnProductionSlot = -not $(Get-AzWebApp -ResourceGroupName $resourceGroupName -Name $appServiceResource.Name -ErrorAction SilentlyContinue).SiteConfig.RemoteDebuggingEnabled
+        
+            # Get all non-production slots for this App Service.
+            $nonProductionSlots = (Get-AzWebAppSlot -ResourceGroupName $resourceGroupName -Name $resourceName) | select -Property Name | ForEach {$_.Name}
+            $nonProductionSlotsWithoutRemoteDebuggingDisabledStr = [String]::Empty
+            $list = @()
+            $IsRemoteDebuggingDisabledOnAllNonProductionSlots = $true
+            $nonProductionSlots | ForEach-Object{
+                $nonProdSlotName = $_
+                $nonProdSlotName = $nonProdSlotName.Split('/')[1]
+                $nonProductionSlotConfigurations = Get-AzWebAppSlot -ResourceGroupName $resourceGroupName -Name $resourceName -Slot $nonProdSlotName
+
+                $isRemoteDebuggingDisabledOnNonProductionSlots = -not $($nonProductionSlotConfigurations.SiteConfig.RemoteDebuggingEnabled -contains $true)
```

Comment:
We can have simple check like: $nonProductionSlotConfigurations.SiteConfig.RemoteDebuggingEnabled -eq $true

Reply:
Done.